### DISCUSSION TITLE
Support for WINDOW clause

### DISCUSSION THREAD

**Discussion by simonnelse** [sqlalchemy/sqlalchemy#9496] [Usage Questions] [answered]
I was wondering if there is support for the [WINDOW](https://www.postgresql.org/docs/current/tutorial-window.html) clause in SQLAlchemy. 

In PostgreSQL, windowing behaviour can be named in a **WINDOW** clause and then referenced in **OVER**. However, I haven't seen this feature in SQLAlchemy.

For example, in PostgreSQL, we can write:

```sql
SELECT sum(salary) OVER w, avg(salary) OVER w
  FROM empsalary
  WINDOW w AS (PARTITION BY depname ORDER BY salary DESC);
```

Is there a way to achieve the same behaviour in SQLAlchemy? Or is this something that could be added in the future?

Thank you!

**Comment by zzzeek** [comment]
hi -

the tutorial has a whole section on window functions:

https://docs.sqlalchemy.org/en/20/tutorial/data_select.html#using-window-functions

**Comment by CaselIT** [comment]
Hi,

This is tracked by this issue https://github.com/sqlalchemy/sqlalchemy/issues/8352

**Reply by zzzeek** [reply]
oh named, got it.  sure

**Reply by simonnelse** [reply, marked as answer]
As of now, I can make use of this workaround - https://github.com/sqlalchemy/sqlalchemy/issues/8352#issuecomment-1205724074 to avoid code duplication. Thank you ❤️.
